quic/s2n-quic-transport/src/recovery/manager.rs

```diff
@@ -304,6 +305,43 @@ impl<Config: endpoint::Config> Manager<Config> {
         self.pto.on_transmit(context)
     }
 
+    /// Process pending ACK information.
+    ///
+    /// Update congestion controller, timers and meta data around acked packet ranges.
+    pub fn on_pending_ack_ranges<Ctx: Context<Config>, Pub: event::ConnectionPublisher>(
```

Comment:
What would you think about converting the PendingAckRanges back into an Ack frame before calling the recovery::Manager? Something like:

```rust
impl From<PendingAckRanges> for Ack<AckRanges> {
    fn from(pending: PendingAckRanges) -> Self {
        Ack {
            ack_delay: unsafe { VarInt::new_unchecked(pending.ack_delay.as_micros() as u64) },
            ack_ranges: pending.ranges,
            ecn_counts: Some(pending.ecn_counts),
        }
    }
}
```
and then just call `on_ack_frame` as normal

I kinda don't really like the recovery::Manager even knowing about pending ack ranges, since it doesn't matter for the recovery manager if they were pending/delayed or just processed immediately. Unless you think the more extreme versions of delayed ack processing would require more data in PendingAckRanges that recovery::Manager will need to know about.

Reply:
That seems to be creating VarInt objects so it doesnt less than ideal.

If a single entry/interface is desired then I propose we move away from ack::Frame and PendingAckRanges and just have something like:

```
on_ack(
  range: impl Iterator<Item = PacketNumberRange>,
  ...
)
```

The disadvantages of the of the above strategy is that we need to pass in more fields individually(max_delay, ecn_counts, largest_pn) and that becomes more cluttered.

Reply:
We could rename PendingAckRanges to AckRangesInfo and have that be the common interface. That would be more ideal since we want to end up with PacketNumber/PacketNumberRange eventually

Reply:
yeah I thought about suggesting the same thing. It's _probably_ clearer to split it out into separate arguments rather than convert it to a frame only to convert it back into the values that they already were, I would think.

Reply:
spoke offline and it would be nice to have a common interface into the recovery/manger; something that takes `impl Iterator<Item = PacketNumberRange>`.

However, since the interface might change and depending on the exact implementation its not obvious that there should be a common function and what it should look like. In order to avoid multiple refactors I am adding a [task](https://github.com/aws/s2n-quic/issues/1277) to come back to this after the implementation is set.